snakemake/script.py

```diff
@@ -611,9 +611,9 @@ def write_script(self, preamble, fd):
 
     def _is_python_env(self):
         if self.conda_env is not None and ON_WINDOWS:
-            prefix = self.conda_env
+            prefix = os.path.join(self.conda_base_path, "envs", self.conda_env)
         elif self.conda_env is not None:
-            prefix = os.path.join(self.conda_env, "bin")
+            prefix = os.path.join(self.conda_base_path, "envs", self.conda_env, "bin")
```

Comment:
Conda envs are not always named. If they are not named (which is actually the recommended default), self.conda_env is already the correct path. Hence, you need to check here whether it is named or not. I wonder how this actually worked in the test cases apart from the prebuilt one. Maybe first try self.conda_env alone, and if that fails, try prefixing it with the conda base path?

Reply:
Sorry for such a long delay.  i have added `if` statements to detect if  `self.conda_env` is a correct path and to add `self.conda_base_path` if necessary.